### PR TITLE
[improve] improve hazelcast joiner, lite node can't be election as master

### LINKED ISSUE #8248: [Bug] [Zeta] Separated Cluster deploy has some issue on K8S
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

I am trying deploy Zeta Cluster 2.3.8 on K8S use Separated Cluster mode.

But I find the deploy order is very important.

When I start `worker` first, the `worker` can be choose as the `hazelcast master`!!! 
image

When `master` node startup, it will join the cluster as the `follower`.

When api or other operation called, it will call `master` sometimes, But the current cluster `master` is a worker node, So the cluster can work well.

 

### SeaTunnel Version

2.3.8

### SeaTunnel Config

```conf
na
```

### Running Command

```shell
na
```

### Error Exception

```log
na
```

### Zeta or Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by liunaijie**
### Purpose of this pull request
close #8248

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by liunaijie** [comment]
overwrite the default behavoir, use overwrite `Joiner`

https://github.com/hazelcast/hazelcast/blob/v5.1/hazelcast/src/main/java/com/hazelcast/instance/impl/Node.java#L841

https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/DiscoveryJoiner.java#L38

https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/MulticastJoiner.java#L34

**Comment by liunaijie** [comment]
tested in my company with K8S deploy, it can work well

**Review by Hisoka-X — COMMENTED** [review]
Could you update https://github.com/apache/seatunnel/blob/1a5feb1dfda8ea91f1f46f18666cd685c028b8d7/seatunnel-engine/seatunnel-engine-client/src/test/java/org/apache/seatunnel/engine/client/SeaTunnelEngineClusterRoleTest.java#L52 to test this situation?

**Comment by Hisoka-X** [comment]
or test it in https://github.com/apache/seatunnel/blob/1a5feb1dfda8ea91f1f46f18666cd685c028b8d7/seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterIT.java#L44

**Comment by liunaijie** [comment]
> Could you update
> 
> https://github.com/apache/seatunnel/blob/1a5feb1dfda8ea91f1f46f18666cd685c028b8d7/seatunnel-engine/seatunnel-engine-client/src/test/java/org/apache/seatunnel/engine/client/SeaTunnelEngineClusterRoleTest.java#L52
> 
> to test this situation?

Add test about TcpIp and multicast.
The K8S can re-use the exist e2e test https://github.com/apache/seatunnel/blob/dev/seatunnel-e2e/seatunnel-engine-e2e/seatunnel-engine-k8s-e2e/src/test/java/org/apache/seatunnel/engine/e2e/k8s/KubernetesIT.java#L72

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/SeaTunnelNodeContext.java [hunk 1]
```diff
@@ -45,15 +50,28 @@ public NodeExtension createNodeExtension(@NonNull Node node) {
...
+        } else if (join.getTcpIpConfig().isEnabled()) {
             log.info("Using LiteNodeDropOutTcpIpJoiner TCP/IP discovery");
             return new LiteNodeDropOutTcpIpJoiner(node);
+        } else if (node.getProperties().getBoolean(DISCOVERY_SPI_ENABLED)
+                || isAnyAliasedConfigEnabled(join)
+                || join.isAutoDetectionEnabled()) {
+            log.info("Using LiteNodeDropOutDiscoveryJoiner Discovery SPI");
+            return new LiteNodeDropOutDiscoveryJoiner(node);
         }
+        return null;
```
Comment (Hisoka-X) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/SeaTunnelNodeContext.java:71]: why not return `super.createJoiner(node);`?

Comment (liunaijie) [reply seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/SeaTunnelNodeContext.java:71]: Has cover all cases. Same like original implement.

image

**Review by Hisoka-X — APPROVED** [review]

**Review by hailin0 — APPROVED** [review]

**Pull request merged**